juju/juju
# Use the pool system state for getting api addresses

When getting controller API addresses, the controller model state is needed for k8s models as that's where the controller cloud service entity is stored. We were constructing a new controller model state for each API call, but that's not needed as the state pool is already available on the facade context, and said state pool has a reference to SystemState().

This PR weaves use of the controller model through the facade layers, for both CAAS and IAAS models.

## QA steps

bootstrap both k8s and lxd controllers
deploy charms
enable ha on lxd controller
check local controller yaml for correct api addresses
juju show-controller and check api addresses
ssh into charm operator and check agent conf for correct api addresses
